### DISCUSSION TITLE
Openstreetmap style

### DISCUSSION THREAD

**Discussion by CATALYST1109** [protomaps/basemaps#52] [Q&A] [answered]
Hi,
I am using the small map downloaded from protomaps [here](https://app.protomaps.com/downloads/small_map). I am using the dark and light styles/layers from the protomap basemap. The documentation says it is latest OSM data, but upon comparing with the OSM [map](https://www.openstreetmap.org/), I can see a lot more info on the OSM map like hospitals,schools,landmarks,local area names and such, which I can't see on the proto-map. I am using maplibregl js to render my proto-map with pmtiles.
So my questions are the following:- 
1) Is the protomaps data somehow different(less detailed) or is it a style thing or something else entirely? 
2) If it is indeed a style thing, is there a way to get and use the open street map tile style or the style used in the OSM map above with protomaps?

Thanks in advance!

**Comment by nvkelso** [comment, marked as answer]
The OSM.org map style is rich in detail and complexity, having been developed over many years for data editors to distinguish a wide range of features. OpenStreetMap has an extremely complicated data schema.

In comparison, the Protomaps vector tile schema is meant to be a "lite" schema focused on small tile size and essential data visualization use case.

The public endpoint has v2 of the Protomaps schema which does well on small tile size but at the expense of not enough features.

Many, but not all, of the gaps you identify are being addressed in https://github.com/protomaps/basemaps/pull/47 and an earlier related PR, for a planned v3 of the schema that more closely aligns it with the open source Tilezen schema and has wider range of data carried over from OSM, including more land use and POI features.

If there are specific gaps please list them here and we can discuss how to address them (if they are already covered in my PR or could be added).

**Reply by CATALYST1109** [reply]
Thanks for taking out the time to reply :) 
Some context --> My use case is vehicle GPS tracking related, which means the map would need to have enough location markers for the user to make out the current location using nearby landmarks and such at a city level. The idea of a static blob rather than a full-on data service or hosted maps API is what pulled me to Protomaps and PMtiles. Not to mention the reduced costs of operation. The area of operation is small, say limited to a state, so the file size remains relatively small.
But the sparsity of location markers (listed below) makes it difficult to make out where the vehicle currently is. Just road names, which I see most of in protomaps, isn't very useful when viewing at a fairly zoomed in level. I don't envision having every cafe and store marked while keeping the tile small, just major landmarks to help get some bearing. 

Below is a list of location markers, sorted by most useful to least (IMO) in figuring out locations within a city. Can also be thought of as markers least likely to change over a course of time. 

1) Area/colony names, industrial areas/big compounds/mills, markets etc
2) Stadiums and Parks -- the local/community ones, not just national ones/wildlife sanctuaries
3) Schools/colleges/universities
4) Hospitals/clinics
5) Places of worship
6) Hotels/motels
7) Courthouse/jails
8)  tourist spots/local attractions -- not sure how it would be identified as a feature or if it can be uniquely categorized at all
9) Banks/ATMs
10) Cafes/stores

If I'm reading your answer correctly, it seems like protomaps schema i.e. the logical organization of data in protomaps is somehow lossy compressing the OSM data to keep lesser features to keep a tile (a fixed size geographical area data) size limited. So is it possible to support variable tile size, or to have no cap on tile size limit at the expense of file size which one can compensate for using a smaller map area ? That might allow for higher features/larger sized tiles in busier areas like cities/towns and such and save space on the whole map file by having a lot less small sized features in areas of less interest like desserts/forests and such? Might introduce issues when stitching dissimilar styles at higher levels of zoom though. Just thinking out loud.

While I understand the need of a planet scale map, I think the flexibility to serve smaller area maps (say a city/district/state) with greater detail can be quite useful for a majority of gps tracking use cases. If not, maybe there exists a cutoff point where we get just enough detail to make it useful at smaller scales/higher zooms while keeping tile size relatively small ?

**Reply by nvkelso** [reply]
Yes, all or very nearly all of those features should be included in Protomaps basemaps vector tile schema v3. Timeline is the next couple of weeks.

While including extra data does increase file size, luckily there are file size savings in some of the existing layers and features (like feature merging) I am implementing that gives us more headroom for these additions. 

NOTE I may need to research this one more:

> Area/colony names, industrial areas/big compounds/mills, markets etc

- Area/colony names I think are `place = neighbourhood` or `place = suburb` in OSM. 
- Industry areas / big compounds will be shaded by industrial landuse, but I may need more information to draw a POI for them
- mills I think would show up now if they are tagged onto a building
- markets will be there I think already in the PR

**Reply by CATALYST1109** [reply]
I checked out OSM wiki, came up with the following :-

- Area/colony names -- yes `place=neighbourhood` is what I meant by area/colony, wasn't sure of the equivalent OSM lingo.  Neighbourhood would be the area within a suburb, so suburb name could be the thing to display at lower zoom levels and neighbourhoods at higher ones. There apparently exists `place=quarter` as well for something in between the two, so for completeness maybe include those as well? The hierarchy seems to be City > Suburb > Quarter > Neighbourhood. [Reference](https://wiki.openstreetmap.org/wiki/Tag:place%3Dneighbourhood)
- Industry areas / big compounds -- given industrial landuse, apparently industrial parks (IT parks, Special economic zones and such) may have a `place=neighbourhood` tag if they are big enough, so that in conjunction with `name=*` (when available) and `industrial=*` tags might help? Also, looks like `man_made=works` can be used to identify factories. `Utility=*` can also be used for industrial areas dedicated to a specific utility like powerplants and such, but might be rarer. Check useful combinations [here](https://wiki.openstreetmap.org/wiki/Tag:landuse%3Dindustrial)

Let me know if I can help :)

**Comment by CATALYST1109** [comment]
@nvkelso , I see the current spec is now 3, was the public endpoint updated to v3 as well ?

**Reply by bdon** [reply]
I'm hoping to get a beta release of v3 on a public endpoint this week - however, this will need to match a corresponding version of the style

**Reply by CATALYST1109** [reply]
Could you please add a compatibility table to the docs so people can know which spec version needs to be paired to which style version ? 
One more suggestion/request - now that you have multiple spec versions, maybe please specify the spec version on the downloads page? I might have missed it , but I tried to check the spec version on the downloads page and could not find any. That would help people upgrade seamlessly.

**Reply by cdavis4** [reply]
Ok. It sounds like the protomaps/basemaps is not backwards compatible. You need to use a version of the build to create pmtiles that will work with a the version of protomaps-themes-base you are using. I'm getting it. A spec pairing doc would be helpful.

**Reply by nvkelso** [reply]
Yes, the PMTiles major version needs to match with the map style. Sorry for confusion as the version `3` spec settles – there are some backwards incompatible changes versus version `2` as you've noted (but also a lot of new features to temp you to upgrade).

@bdon and I will confer about how to present and maintain this paired versioning going forward.

**Reply by bdon** [reply]
hey all, some major updates coming:

* https://maps.protomaps.com/ points at a daily build of the planet.pmtiles
* there's a select box at the top that lets you look at a published version of the NPM packages, I just put out `2.0.0-alpha.0` which is compatible with tileset v3.0

(screenshot)

<img width="955" alt="Screenshot 2023-09-21 at 15 03 50" src="https://github.com/protomaps/basemaps/assets/77501/a3ec82a9-6d5f-4769-9a6d-75a03df6df70">
